app/src/main/java/com/eveningoutpost/dexdrip/StopSensor.java

```diff
@@ -48,12 +48,33 @@ public void addListenerOnButton() {
 
         button.setOnClickListener(new View.OnClickListener() {
             public void onClick(View v) {
+                confirm_stop();
+            }
+
+        });
+    }
+
+    public void confirm_stop() {
+        final AlertDialog.Builder build_stop = new AlertDialog.Builder(this);
+        build_stop.setTitle(gs(R.string.are_you_sure));
+        build_stop.setMessage(gs(R.string.do_you_want_to_end_session));
```

Comment:
I suggest naming the resource string `confirm_sensor_stop`.

Reply:
Or `sensor_confirm_stop`. A naming scheme for resource strings would be helpful for translators to understand the context.

Reply:
> @tolot27 The only new resource string in this PR is the following.
> do_you_want_to_end_session
> 
> confirm_stop is not a resource string. Am I misunderstanding you? Please explain.

I thought it was obvious which resource string I meant because I created the comment at line 60 and only one new resource string was introduced. Try to improve this in the future.

Yes, I suggest to rename `do_you_want_to_end_session` into `confirm_sensor_stop` or better into `sensor_confirm_stop` because the previous name is out of context. Meanwhile, this PR should just address the missing resource string in 84e4376.

Reply:
Your suggestion reads: I suggest naming the resource string confirm_sensor_stop.  And you have linked to the existing confirm_stop() in StopSensor.java.

Is confirm_stop() is a resource string?
I thought resource strings are in strings.xml.

I want to understand what you are suggesting so that I can do it right in the future.  Asking me to do it right in the future is pointless and humiliating for no reason.

Reply:
The resource string for confirming clear calibration is:
do_you_want_to_delete_and_reset_the_calibrations_for_this_sensor

I set the one for asking for confirmation to stop to the following based on that.
do_you_want_to_end_session

Because I thought similar to the calibration, it reflects the sentence that should be translated.

Do you think that the existing resource string for calibration is also bad?
If not, I don't see how confirm_sensor_stop represents the sentence that should be translated.

You can point me to documentation that explains what you need.
Or, you can tell me what you need.  In that case, you should be open to questions coming from me.
If you don't have time, there is no rush.  Your feedbcak is always welcome.

Saying do better next time is not useful.

Reply:
`confirm_stop()` is a Java function you have added. But you also added the line `build_stop.setMessage(gs(R.string.do_you_want_to_end_session));`. With `gs(R.string.do_you_want_to_end_session)` the translated resource string with name `do_you_want_to_end_session` is retrieved from the AppContext. As you can see, every resource string is prefixed with `R.string.`.

> I thought resource strings are in strings.xml.

You have to specify it in strings.xml and you reference it where you need it. It does not matter where you rename the string (in Java or xml file) as long as you are using code refactoring. This ensures, it is renamed at every place of use.

Reply:
> Saying do better next time is not useful.

Where did I say this? I tried to explain my suggestion with "A naming scheme for resource strings would be helpful for translators to understand the context."

Reply:
I followed the existing resource strings, as examples, to figure out what should be acceptable.  
If what I have done is not good, I welcome directions.  
For example, you can say, the resource string should match the entire sentence.  Or, you can say, the resource string must have exactly 12 words, or, the resource string must ...

I still don't know what is wrong with my resource string.
And, I have no choice but to follow the existing examples unless you give me clear directions.

Reply:
![Screenshot 2021-05-16 155410](https://user-images.githubusercontent.com/51497406/118410755-1d05cd80-b65f-11eb-9d1d-dcdde2e8e88f.png)

Reply:
![Screenshot 2021-05-16 155410](https://user-images.githubusercontent.com/51497406/118410912-e11f3800-b65f-11eb-8956-820cc76b35dc.png)

Reply:
The resource string should not repeat the sentence which is shown (without cutting some words). A resource string should contain the context, i. e. my suggestion `confirm_sensor_stop` or `sensor_confirm_stop`. It is short, it is like a label, it does not repeat the text to translate, it contains the context "sensor", a verb "confirm" and an "action". The suggested `sensor_confirm_stop` is according some (currently un-) defined nomenclature containing the class name (`Sensor`, at least partially) and method of use (`confirm_stop()`). Hence, the context (of use) is obvious and the string is meaningful for translators, as the Java function is.
My suggestion is to be precise and provide context.

Reply:
I follow that.  
So, the existing resource string for the calibration confirmation also does not follow your requirement, right?
If that's correct, I understand.